### ISSUE TITLE
Add more options for searching

### ISSUE THREAD

**Issue by stopyoukid** [Microsoft/PowerBI-visuals-AttributeSlicer#25]
- Starts With
- Ends With
- Contains
...

**Comment by JonJagd** [comment]
Hi

Thanks for the great visual Attribute Slicer. I am using it quite a lot when needing to give my Power BI users an efficient and fast performing search function, to help them select a value from a long list of values. However we have one issue that I have not found a solution to.  It looks like the search works with a 'like' operator, so if I search for the unique value 123 it does a "like %123%" which returns a long list of values that starts with, contains and ends with 123. That sort of cancels the idea of using it as a search field, at least in our case.

Is it possible to tweak it to do an exact match? Sort of like using an EQUALS operator instead of a LIKE?

Br,
Jon

**Comment by stopyoukid** [comment]
Yes, it can be tweaked.  The lines you will be interested in are:

https://github.com/Microsoft/PowerBI-visuals-AttributeSlicer/blob/develop/packages/attribute-slicer-powerbi/src/state.ts#L68

The `buildContainsFilter` function is where is actually generates the `contains` filter, and that comes from here: https://github.com/Microsoft/Essex-PowerBI-visuals-base/blob/master/src/lib/utils/buildContainsFilter.ts#L33-L62

You could copy that function into your repo and tweak it.  The line you would need to change is this one: https://github.com/Microsoft/Essex-PowerBI-visuals-base/blob/master/src/lib/utils/buildContainsFilter.ts#L44

On that line, you can see it is creating a `contains` expression, if you change that to `equal`, it should work as you require.

You will also need to change this line to use your new function: https://github.com/Microsoft/PowerBI-visuals-AttributeSlicer/blob/develop/packages/attribute-slicer-powerbi/src/AttributeSlicerVisual.ts#L462

**Comment by JonJagd** [comment]
Fantastic, I will definitely try that. Thanks a lot for your thorough and detailed answer.

**Comment by wwoodbery** [comment]
Hi everyone,

Hoping someone is still out there who will read this.

I too would like a search visual that does unique/exact search, but I'm quite new at this, so I'm finding it very difficult to get my `pbiviz package` to work. 

This search tweak seems like a very easy thing for someone who knows what they're doing to implement. So, I would be **SUPER** grateful if someone could implement this and share the finished/packaged .pbiviz file.

Best,
Wyatt
